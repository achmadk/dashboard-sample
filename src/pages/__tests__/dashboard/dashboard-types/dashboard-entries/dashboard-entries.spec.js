import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import chai from 'chai'

import {
  DashboardEntries,
  mapStateToProps
} from '@/pages/dashboard/dashboard-types/dashboard-entries/dashboard-entries'

describe('Dashboard Entries Component', () => {
  it('render correctly', () => {
    const props = {
      match: {
        url: '/dashboard/average_tender_duration'
      }
    }
    const wrapper = shallow(<DashboardEntries {...props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('render correctly without matching URL from dashboard-contents', () => {
    const props = {
      match: {
        url: '/dashboard/average_tender_durations'
      }
    }
    const wrapper = shallow(<DashboardEntries {...props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('return userParams value', () => {
    const state = {
      user: {
        data: {
          username: 'chack',
          authn_token: 'check'
        }
      }
    }
    const result = mapStateToProps(state)
    chai.expect(result).to.be.not.undefined
  })
})
