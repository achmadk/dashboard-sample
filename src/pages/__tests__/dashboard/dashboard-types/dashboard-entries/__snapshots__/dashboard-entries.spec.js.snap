// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard Entries Component render correctly 1`] = `
<Fragment>
  <h2
    className="article-title"
  />
  <div
    className="ui-animate"
  >
    <WithStyles(LoadingComponent) />
    <pre>
      null
    </pre>
  </div>
</Fragment>
`;

exports[`Dashboard Entries Component render correctly without matching URL from dashboard-contents 1`] = `
<Fragment>
  <h2
    className="article-title"
  />
  <div
    className="ui-animate"
  >
    <WithStyles(LoadingComponent) />
    <pre>
      null
    </pre>
  </div>
</Fragment>
`;
